# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationUtils.h
# Commit: bde8ca8fd963
# Full Hash: bde8ca8fd9636bc6650ee6768dfb8e60fc1da596
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 6: Move NotificationObserver functions to NotificationUtils r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227691
# ==============================================================================

diff -r fca3921d09a5 -r bde8ca8fd963 dom/notification/NotificationUtils.h
--- a/dom/notification/NotificationUtils.h	Mon Nov 11 20:34:13 2024 +0000
+++ b/dom/notification/NotificationUtils.h	Mon Nov 11 20:34:13 2024 +0000
@@ -84,6 +84,13 @@
 void UnregisterNotification(nsIPrincipal* aPrincipal, const nsString& aId,
                             const nsString& aAlertName, CloseMode aCloseMode);
 
+nsresult RemovePermission(nsIPrincipal* aPrincipal);
+nsresult OpenSettings(nsIPrincipal* aPrincipal);
+
+enum class NotificationStatusChange { Shown, Closed };
+nsresult AdjustPushQuota(nsIPrincipal* aPrincipal,
+                         NotificationStatusChange aChange);
+
 }  // namespace mozilla::dom::notification
 
 #endif  // DOM_NOTIFICATION_NOTIFICATIONUTILS_H_